clash-lib/src/Clash/Normalize/Transformations/Specialize.hs

```diff
@@ -488,6 +500,18 @@ specialize' (TransformContext is0 _) e (Var f, args, ticks) specArgIn = do
     noUserInline = NoUserInline
 #endif
 
+    specializeName :: (InlineSpec, Name Term) -> (Maybe InlineSpec, Name Term) -> Name Term
+    specializeName (spec0, n0) (Just spec1, n1)
+      | isNoInline spec0
+      , isNoInline spec1
+      = n0{nameOcc=nameOcc n0 <> "_" <> stripMods (nameOcc n1)}
+      | isNoInline spec1
+      = n1
+    specializeName (_, n0) _ = n0
+
+    stripMods :: Text.Text -> Text.Text
+    stripMods = Text.takeWhileEnd (/= '.')
```

Comment:
This doesn't work when e.g. have `Data.Bits..&.`

Reply:
Perhaps we should just add another issue, and leave this as is for now. We might want to refactor names so module name are stored separately.